src/coordination/coordinator_state_manager.cpp

```diff
@@ -93,35 +94,57 @@ auto CoordinatorStateManager::HandleVersionMigration() -> void {
         "and run queries to add instances and coordinators to cluster.");
   }
 }
-CoordinatorStateManager::CoordinatorStateManager(CoordinatorStateManagerConfig const &config, LoggerWrapper logger)
+CoordinatorStateManager::CoordinatorStateManager(CoordinatorStateManagerConfig const &config, LoggerWrapper logger,
+                                                 std::optional<CoordinationClusterChangeObserver> observer)
     : my_id_(static_cast<int>(config.coordinator_id_)),
       cur_log_store_(cs_new<CoordinatorLogStore>(logger, config.log_store_durability_)),
       logger_(logger),
-      durability_(config.state_manager_durability_dir_) {
+      durability_(config.state_manager_durability_dir_),
+      observer_(observer) {
   auto const c2c = CoordinatorToCoordinatorConfig{
+      // TODO(antonio) sync with Andi on this one
       config.coordinator_id_, io::network::Endpoint("0.0.0.0", config.bolt_port_),
       io::network::Endpoint{"0.0.0.0", static_cast<uint16_t>(config.coordinator_port_)},
-      io::network::Endpoint{"0.0.0.0", static_cast<uint16_t>(config.management_port_)}, config.coordinator_hostname};
+      io::network::Endpoint{config.coordinator_hostname, static_cast<uint16_t>(config.management_port_)},
```

Comment:
not 100% sure, check how coordinators are returned in show instances and then see if this change makes sense. With this change yo for sure need to change that part of the code so just like this isn't good

Reply:
GetCoordinatorsInstanceStatus this method

Reply:
resolved on call